### PR TITLE
feat: handle empty package.json dependencies better

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
If the package.json has no `dependencies` property defined then assume no deps and do not throw and error that `node_modules` are needed.

Example **package.json**:
```
{
  "name": "subpkg",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}
```

Some lerna projects may have many sub packages that have no dependencies and scanning them with `--all-projects` skips the local packages & throws an error.

#### Additional questions
**Before**
Screen Shot 2020-08-28 at 16 58 09
**After**
Screen Shot 2020-08-28 at 16 57 10

### CODE REVIEW

**Review by gitphill — COMMENTED** [review]
covered by existing tests?

File: src/lib/plugins/nodejs-plugin/npm-modules-parser.ts [hunk 1]
```diff
@@ -1,21 +1,62 @@
...
 import * as resolveNodeDeps from 'snyk-resolve-deps';
+import * as baseDebug from 'debug';
+import * as _ from '@snyk/lodash';
 
 import * as spinner from '../../spinner';
 import * as analytics from '../../analytics';
 import { Options } from '../types';
+import { getFileContents } from '../../get-file-contents';
+
+const debug = baseDebug('snyk-test');
```
Comment (gitphill) [inline src/lib/plugins/nodejs-plugin/npm-modules-parser.ts:12]: snyk-nodejs-plugin?

File: src/lib/get-file-contents.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+export function getFileContents(
+  root: string,
+  fileName: string,
+): {
+  content: string;
+  name: string;
```
Comment (gitphill) [inline src/lib/get-file-contents.ts:9]: filename?

File: src/lib/plugins/nodejs-plugin/npm-modules-parser.ts [hunk 3]
```diff
@@ -1,21 +1,62 @@
 import * as path from 'path';
 import * as fs from 'fs';
 import * as resolveNodeDeps from 'snyk-resolve-deps';
+import * as baseDebug from 'debug';
+import * as _ from '@snyk/lodash';
```
Comment (gitphill) [inline src/lib/plugins/nodejs-plugin/npm-modules-parser.ts:5]: are we still using this fork?

File: test/fixtures/npm/no-dependencies/package.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "custom-name-package",
+  "version": "0.6.6",
+  "description": "",
```
Comment (gitphill) [inline test/fixtures/npm/no-dependencies/package.json:4]: no dependencies?

**Review by gitphill — APPROVED** [review]

File: test/fixtures/npm/no-dependencies/package.json (hunk 4 above)
Comment (lili2311) [reply test/fixtures/npm/no-dependencies/package.json:4]: correct that is the test, empty or not defined should not fail everything

File: src/lib/plugins/nodejs-plugin/npm-modules-parser.ts (hunk 3 above)
Comment (lili2311) [reply src/lib/plugins/nodejs-plugin/npm-modules-parser.ts:5]: yeah we only still use this, haven't moved back yet it seems

**Pull request merged**